### DISCUSSION TITLE
How i create chinees or arabic text with imgaemagick

### DISCUSSION THREAD

**Discussion by zaib347** [dlemstra/Magick.NET#898] [General]
i am not able to add text with other languages 

             
 `string test = "حقوق";
                string caption = $"label: {test } ";
                using (MagickImage watermark = new MagickImage(caption, textSettings))
                {
                    watermark.Rotate(options.TextOptions.RotationAngle);
                    watermark.Extent(watermark.Width + options.TextOptions.PaddingX, watermark.Height + options.TextOptions.PaddingY, 
                    Gravity.Center, MagickColors.Transparent);
                    watermark.Evaluate(Channels.Alpha, EvaluateOperator.Divide, options.TextOptions.Opacity);
                    var imageGravity = Functions.GetMagickGravity(options.TextOptions.position);
                    image.Composite(watermark, imageGravity , CompositeOperator.Over);
                    imgPerview.Source = image.ToBitmapSource();
                }`

**Comment by dlemstra** [comment]
You wil need to use a font that has support for reading those characters.

**Reply by zaib347** [reply]
i am using text  حقوق and fontfamilt Times New Roman text direction right to left and mu output is ح ق وق

**Reply by dlemstra** [reply]
The freetype font libary that is used with label is probably not the best library for Arabic fonts. I would advise you to use pango instead:

```C#
string test = "حقوق";
string caption = $"pango:{test}";
var readSettings = new MagickReadSettings()
{
    Font = "Times New Roman",
};

using (MagickImage watermark = new MagickImage(caption, readSettings))
```
